### PR TITLE
Add initial content for the repo

### LINKED ISSUE #1: Create Helm Resources for WSO2 Micro Integrator version 1.2.0
**Description:**
Add initial content for the repository.

### PR DESCRIPTION
**Pull request by ThilinaManamgoda**
Client: &version.Version{SemVer:"v2.14.1", GitCommit:"5270352a09c7e8b6e8c9593002a73535276507c0", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.1", GitCommit:"5270352a09c7e8b6e8c9593002a73535276507c0", GitTreeState:"clean"}

### CODE REVIEW

**Comment by Aaquiff** [comment]
@ThilinaManamgoda if possible, can we template hostname in the values.yaml instead of using the release name?

File: helm/micro-integrator/README.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+* In order to use WSO2 Helm resources, you need an active WSO2 subscription. If you do not possess an active WSO2
+  subscription already, you can sign up for a WSO2 Free Trial Subscription from [here](https://wso2.com/free-trial-subscription)
+  . Otherwise you can proceed with docker images which are created using GA releases.<br><br>
+
+* Install [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), [Helm](https://github.com/kubernetes/helm/blob/master/docs/install.md)
+(and Tiller) and [Kubernetes client](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (compatible with v1.10) in order to run the 
+steps provided in the following quick start guide.<br><br>
+
+* An already setup [Kubernetes cluster](https://kubernetes.io/docs/setup/pick-right-solution/).<br><br>
```
Comment (Aaquiff) [inline helm/micro-integrator/README.md:18]: Broken link, we can use **https://kubernetes.io/docs/setup/** instead

File: helm/micro-integrator/README.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+* [Quick Start Guide](#quick-start-guide)
+
+## Prerequisites
+
+* In order to use WSO2 Helm resources, you need an active WSO2 subscription. If you do not possess an active WSO2
+  subscription already, you can sign up for a WSO2 Free Trial Subscription from [here](https://wso2.com/free-trial-subscription)
+  . Otherwise you can proceed with docker images which are created using GA releases.<br><br>
+
+* Install [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), [Helm](https://github.com/kubernetes/helm/blob/master/docs/install.md)
+(and Tiller) and [Kubernetes client](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (compatible with v1.10) in order to run the
```
Comment (Aaquiff) [inline helm/micro-integrator/README.md:15]: Can we use a newer version of kubectl?

File: helm/micro-integrator/README.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  . Otherwise you can proceed with docker images which are created using GA releases.<br><br>
+
+* Install [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), [Helm](https://github.com/kubernetes/helm/blob/master/docs/install.md)
+(and Tiller) and [Kubernetes client](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (compatible with v1.10) in order to run the 
+steps provided in the following quick start guide.<br><br>
+
+* An already setup [Kubernetes cluster](https://kubernetes.io/docs/setup/pick-right-solution/).<br><br>
+
+* Install [NGINX Ingress Controller](https://kubernetes.github.io/ingress-nginx/deploy/). Please note that Helm resources for WSO2 product
+deployment patterns are compatible with NGINX Ingress Controller Git release [`nginx-0.22.0`](https://github.com/kubernetes/ingress-nginx/releases/tag/nginx-0.22.0).
```
Comment (Aaquiff) [inline helm/micro-integrator/README.md:21]: We no longer need to 0.22.0 of NGINX right?

File: helm/micro-integrator/templates/micro-integrator-conf.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Aaquiff) [inline helm/micro-integrator/templates/micro-integrator-conf.yaml:1]: Update license header

File: helm/micro-integrator/templates/NOTES.txt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+Thank you for installing WSO2 Micro Integrator.
+
+Please follow these steps to assess Micro Integrator.
+
+1. Obtain the external IP (`EXTERNAL-IP`) of the Ingress resources by listing down the Kubernetes Ingresses.
+
+  kubectl get ing -n {{ .Release.Namespace }}
+
+    NAME                       HOSTS                     ADDRESS        PORTS     AGE
+    {{ include "micro-integrator.fullname" . }}       {{ .Release.Name }}       <EXTERNAL-IP>  80, 443   3m
```
Comment (Aaquiff) [inline helm/micro-integrator/templates/NOTES.txt:10]: @ThilinaManamgoda hostname should be changed here as well

File: helm/micro-integrator/templates/NOTES.txt [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+2. Add the above host as an entry in /etc/hosts file as follows:
+
+    <EXTERNAL-IP>	{{ .Release.Name }}
```
Comment (Aaquiff) [inline helm/micro-integrator/templates/NOTES.txt:14]: @ThilinaManamgoda hostname should be changed here as well

File: helm/micro-integrator/README.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+* [Prerequisites](#prerequisites)
+* [Quick Start Guide](#quick-start-guide)
+
+## Prerequisites
+
+* In order to use WSO2 Helm resources, you need an active WSO2 subscription. If you do not possess an active WSO2
+  subscription already, you can sign up for a WSO2 Free Trial Subscription from [here](https://wso2.com/free-trial-subscription)
+  . Otherwise you can proceed with docker images which are created using GA releases.<br><br>
+
+* Install [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), [Helm](https://github.com/kubernetes/helm/blob/master/docs/install.md)
```
Comment (Aaquiff) [inline helm/micro-integrator/README.md:14]: Helm link is broken. Shall we use **https://helm.sh/docs/intro/install/**

File: helm/micro-integrator/README.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+(and Tiller) and [Kubernetes client](https://kubernetes.io/docs/tasks/tools/install-kubectl/) in order to run the 
+steps provided in the following quick start guide.<br><br>
+
+* An already setup [Kubernetes cluster](https://kubernetes.io/docs/setup/pick-right-solution/).<br><br>
+
+* Install [NGINX Ingress Controller](https://kubernetes.io/docs/setup/). Please note that Helm resources for WSO2 product
```
Comment (Aaquiff) [inline helm/micro-integrator/README.md:20]: Ingress Controller link is incorrect.
**https://kubernetes.github.io/ingress-nginx/deploy/**

File: helm/micro-integrator/README.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+deployment patterns are compatible with NGINX Ingress Controller Git release [`nginx-0.30.0`](https://github.com
+/kubernetes/ingress-nginx/releases/tag/nginx-0.30.0).
```
Comment (Aaquiff) [inline helm/micro-integrator/README.md:22]: Space in link

File: helm/micro-integrator/templates/logstash-elasticsearch-creds.yaml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+# limitations under the License.
+{{ if .Values.wso2.centralizedLogging.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+    name: logstash-elasticsearch-creds
+    namespace : {{ .Release.Namespace }}
+type: Opaque
+data:
+    username: {{ default "elastic" .Values.wso2.centralizedLogging.logstash.elasticsearch.username | b64enc }}
```
Comment (chirangaalwis) [inline helm/micro-integrator/templates/logstash-elasticsearch-creds.yaml:22]: @ThilinaManamgoda shall we avoid setting default values in the resource definitions and set default values in the `values.yaml` itself?

Comment (chirangaalwis) [reply helm/micro-integrator/templates/logstash-elasticsearch-creds.yaml:22]: Please check all applicable cases.

File: helm/micro-integrator/values.yaml [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+      envs:
+  # Configure Ingress for the Micro integrator deployment.
+  ingress:
+    # Hostname for the Micro Integrator.
+    host:
+    # Annotations for the Micro Integrator Ingress.
+    annotations:
+      ingress.kubernetes.io/ssl-redirect: 'true'
+      kubernetes.io/ingress.class: "nginx"
+      nginx.ingress.kubernetes.io/affinity: "cookie"
```
Comment (chirangaalwis) [inline helm/micro-integrator/values.yaml:70]: @ThilinaManamgoda just out of curiosity, do we need session affinity in Micro Integrator?

Comment (ThilinaManamgoda) [reply helm/micro-integrator/values.yaml:70]: It is not needed. I'll remove this.

File: helm/micro-integrator/templates/logstash-elasticsearch-creds.yaml (hunk 10 above)
Comment (ThilinaManamgoda) [reply helm/micro-integrator/templates/logstash-elasticsearch-creds.yaml]: Done

**Pull request merged**